api/src/main/java/org/openmrs/api/impl/AdministrationServiceImpl.java

```diff
@@ -229,16 +285,24 @@ public List<GlobalProperty> getGlobalPropertiesByPrefix(String prefix) {
 	@Override
 	@Transactional(readOnly = true)
 	public List<GlobalProperty> getGlobalPropertiesBySuffix(String suffix) {
-		return dao.getGlobalPropertiesBySuffix(suffix);
+		return filterGlobalPropertiesByViewPrivilege(dao.getGlobalPropertiesBySuffix(suffix));
 	}
 	
 	/**
 	 * @see org.openmrs.api.AdministrationService#purgeGlobalProperty(org.openmrs.GlobalProperty)
 	 */
 	@Override
 	public void purgeGlobalProperty(GlobalProperty globalProperty) throws APIException {
-		notifyGlobalPropertyDelete(globalProperty.getProperty());
-		dao.deleteGlobalProperty(globalProperty);
+		if (globalProperty != null) {
```

Comment:
Since the original code assumed that `globalProperty` is not equal to null, let us also remove the null check.

Reply:
@dkayiwa sure, I have reverted